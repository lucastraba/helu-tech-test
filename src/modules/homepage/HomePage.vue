<template>
  <div class="home-page">
    <NavBar />
    <MainTable />
  </div>
</template>

<script>
import MainTable from '@/modules/covidTable/CovidTable';
import NavBar from '@/modules/ui/NavBar';

export default {
  name: 'HomePage',
  components: {
    NavBar,
    MainTable
  },
  data () {
    return {
      searchTerm: null
    };
  }
};
</script>
