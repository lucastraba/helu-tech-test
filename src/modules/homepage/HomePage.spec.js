import { shallowMount } from '@vue/test-utils';
import HomePage from './HomePage';

describe('NavBar Component', () => {
  it('should render the homepage', () => {
    // Arrange.
    const wrapper = shallowMount(HomePage, {
      stubs: {
        NavBar: true,
        MainTable: true
      }
    });
    // Act.
    // Assert.
    expect(wrapper.find('navbar-stub').exists()).toBe(true);
    expect(wrapper.find('maintable-stub').exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
