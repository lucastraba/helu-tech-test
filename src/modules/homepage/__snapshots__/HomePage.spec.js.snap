// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should render the homepage 1`] = `
"<div class=\\"home-page\\">
  <navbar-stub></navbar-stub>
  <maintable-stub></maintable-stub>
</div>"
`;
